src/web/js-interop.md

```diff
@@ -22,3 +22,110 @@ For help using the `js` package, see the following:
+However, the features available for preview are much
+closer to future JS interop than any pattern supported today.
+So, there are a few reasons to try them out now:
+
+* New JS interop developers can learn and build with future JS interop
+so they won't have to unlearn obsolete patterns in a few months.
+* Existing JS interop developers eager to experiment with
+the latest features in JS interop
+or with `dart2wasm` when it becomes available.
+* Potentially ease transition of existing JS
+interop code once migration becomes necessary.
+
+The following sections are the set of features
+expected to work across compilers for JS interop.
+
+*Requirements:*
+* Dart SDK constraint: `>= 2.19`
+* [`package:js`][] constraint: `>= 0.6.6`
+
+[`dart2wasm`]: https://github.com/dart-lang/sdk/blob/main/pkg/dart2wasm/dart2wasm.md#running-dart2wasm
+[Dart 3]: https://medium.com/dartlang/the-road-to-dart-3-afdd580fbefa
+[`package:js`]: {{site.pub-pkg}}/js
+
+### `package:js`
+
+The key feature of next-generation JS interop is [static interop][].
+We recommend using static interop as the default for `package:js`,
+as it is more declaritive, more likely to be optimized,
+more likely to perform better, and required for `dart2wasm`.
+Static interop addresses several gaps in the existing JS interop story:
+
+* **Missing features:** Static interop enables previously
+unavailable features, like easily wrapping and transforming APIs,
+renaming members, and static checking.
+
+* **Inconsistencies:** Static interop makes backends more consistent,
+so development and production web compilers won't behave as differently
+as before.
+
+* **Clarity:** Static interop takes a step towards making JS interop
+more idiomatic in Dart, and making the boundary between the two languages more visible.
+For example, it enforces that JS classes are not meant to be mixed with Dart
+(e.g. no dynamic calls, JS interop types cannot be implemented as a Dart class).
+We expect upcoming integration with inline classes
+will help make interop even more idiomatic.
+
+You can implement static interop using the `package:js`
+annotation `@staticInterop`.
+The set of features for future static interop currently includes:
+* `@staticInterop` interfaces
+  * External factory constructors with and without `@anonymous`
+  * External `static` class members
+  * External non-`static` extension members on a `@staticInterop`
+  class (fields, getters, setters, methods)
+  * Non-external extension members on a `@staticInterop` class
+* Top-level external members
+* [`@JSExport`][] for mocking and exports
+
+To learn how to implement static interop and see examples,
+visit the [static interop][] specification.
+
+[static interop]: https://pub.dev/packages/js#staticinterop
+[`@JSExport`]: https://pub.dev/packages/js#jsexport-and-js_utilcreatedartexport
+
+
+### `dart:js_util`
+
+We recommend using static interop over `js_util`
+in future JS interop. 
+
+However, since [`js_util`][] provides lower-level
```

Comment:
Reading this again, I am starting to feel the discussion here around why we still have js_util and it's use for migrations could be inadvertently encouraging more usage than we want. So I'm reconsidering whether we should highlight it as much.

For example, how do you feel about replacing this section entirely with a more direct paragraph along these lines:

> `dart:js_util` provides a low-level interop API and is supported by the JS and `dart2wasm` backends. `dart:js_util` can provide more flexibility, for example in rare edge cases we haven't accounted for where static interop is not expressive enough. That said, it is not as ergonomic, and it is not planned to be optimized in the same way as static interop. As a result, we highly recommend using static interop over `dart:js_util` whenever it's possible.

Reply:
That's much better, I agree the last version was a little too encouraging for js_util. Thanks for the text!